eth/client.go

```diff
@@ -548,6 +550,11 @@ func (c *Client) SubscribeToJobEvent(ctx context.Context, logsCh chan types.Log,
 		}
 	}
 
+	q = ethereum.FilterQuery{
```

Comment:
I think we could change this function to:

```
if !IsNullAddress(broadcasterAddr) {
    ....
} else {
   ....
}
```

and include the creation of the filter that uses the broadcasterAddr as the indexed arg in the if block and include the creation of this filter that does not use any indexed args in the else block so we can still use the function for broadcasters that are listening for NewJob events that include a particular broadcaster address